datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -670,19 +687,84 @@ def _submit_gauges_from_histogram(self, metric_name, metric, scraper_config, hos
+                    sample[self.SAMPLE_LABELS]["le"] = str(float(sample[self.SAMPLE_LABELS]["le"]))
+                    tags = self._metric_tags(metric_name, val, sample, scraper_config, hostname)
+                    self.gauge(
+                        "{}.{}.count".format(scraper_config['namespace'], metric_name),
+                        val,
+                        tags=tags,
+                        hostname=custom_hostname,
+                    )
+
+    def _decumulate_histogram_buckets(self, metric):
+        """
+        Decumulate buckets in a given histogram metric and adds the lower_bound label (le being upper_bound)
+        """
+        bucket_values_by_upper_bound = {}
+        for sample in metric.samples:
+            if sample[self.SAMPLE_NAME].endswith("_bucket"):
+                bucket_values_by_upper_bound[float(sample[self.SAMPLE_LABELS]["le"])] = sample[self.SAMPLE_VALUE]
+
+        sorted_buckets = sorted(bucket_values_by_upper_bound.keys())
+
+        # Tuples (lower_bound, upper_bound, value)
+        bucket_tuples_by_upper_bound = {}
+        for i, upper_b in enumerate(sorted_buckets):
+            if i == 0:
+                if upper_b > 0:
+                    # positive buckets start at zero
+                    bucket_tuples_by_upper_bound[upper_b] = (0, upper_b, bucket_values_by_upper_bound[upper_b])
+                else:
+                    # negative buckets start at -inf
+                    bucket_tuples_by_upper_bound[upper_b] = (
+                        float("-inf"),
+                        upper_b,
+                        bucket_values_by_upper_bound[upper_b],
+                    )
+                continue
+            tmp = bucket_values_by_upper_bound[upper_b] - bucket_values_by_upper_bound[sorted_buckets[i - 1]]
+            bucket_tuples_by_upper_bound[upper_b] = (sorted_buckets[i - 1], upper_b, tmp)
+
+        # modify original metric to inject lower_bound & modified value
+        for i, sample in enumerate(metric.samples):
+            if not sample[self.SAMPLE_NAME].endswith("_bucket"):
+                continue
+
+            matching_bucket_tuple = bucket_tuples_by_upper_bound[float(sample[self.SAMPLE_LABELS]["le"])]
+            # Replacing the sample tuple
+            sample[self.SAMPLE_LABELS]["lower_bound"] = str(matching_bucket_tuple[0])
+            metric.samples[i] = (sample[self.SAMPLE_NAME], sample[self.SAMPLE_LABELS], matching_bucket_tuple[2])
+
+    def _submit_sample_histogram_buckets(self, metric_name, sample, scraper_config, hostname=None):
+        if "lower_bound" not in sample[self.SAMPLE_LABELS] or "le" not in sample[self.SAMPLE_LABELS]:
+            self.log.warning(
+                "Metric: {} was not containing required bucket boundaries labels: {}".format(
+                    metric_name, sample[self.SAMPLE_LABELS]
                 )
+            )
+            return
+        sample[self.SAMPLE_LABELS]["le"] = str(float(sample[self.SAMPLE_LABELS]["le"]))
```

Comment:
what's the `str(float(...))` for?

Reply:
Yeah actually it was already done in https://github.com/DataDog/integrations-core/pull/4157/files#r308707128 and I think as pointed out the main reason is to convert is to avoid scientific notation for "small" values like `3.1104e+07` into `31104000.0`